.github/workflows/pytest.yml

```diff
@@ -0,0 +1,192 @@
+          # Needed because release/non-release downloads are different structure
+          cd hsve && if [ ! -f *.whl ]; then mv */*.whl ./; fi
+
+      - name: Download howso-synthesizer-py
+        if: contains(inputs.upstream-details, 'howso-synthesizer-py')
+        env:
+          GH_TOKEN: ${{ secrets.HOWSOAI_WORKFLOW_AUTOMATION_CLASSIC_TOKEN }}
+        run: |
+          run_type=$(echo "${{ inputs.upstream-details }}" | jq '".howso-synthesizer-py"."run-type"')
+          run_id=$(echo "${{ inputs.upstream-details }}" | jq '".howso-synthesizer-py"."run-id"')
+          gh $run_type download -D synth -R "howsoai/howso-synthesizer-py" -p "howso_synthesizer-*-py3-none-any*" "$run_id"
+          # Needed because release/non-release downloads are different structure
+          cd synth && if [ ! -f *.whl ]; then mv */*.whl ./; fi
+
+      - name: Configure MacOS environment
+        if: inputs.platform-pretty == 'MacOS'
+        run: |
+          brew install libomp
+
+      - name: Set up Python
+        uses: actions/setup-python@v4
+        with:
+          python-version: ${{ inputs.python-version }}
+
+      - name: Download tz data
+        if: inputs.platform-pretty == 'Windows'
+        shell: pwsh
+        run: |
+          ./build/powershell/Download-Tzdata.ps1
+
+      - name: Run unit tests
+        run: |
+          python --version
+          if [[ -n "${{ secrets.ARTIFACTORY_ACCESS_TOKEN }}"" ]]; then
+            export PIP_EXTRA_INDEX_URL=https://vsts-build@diveplane.com:${{ secrets.ARTIFACTORY_ACCESS_TOKEN }}@dpbuild.jfrog.io/artifactory/api/pypi/pypi-edge/simple
+          fi
+          if [[ -n "${{ inputs.config-fp }}"]]; then
+            export HOWSO_CONFIG=${{ inputs.config-fp }}
+          fi
+          pip install -r requirements-${{ inputs.python-version }}.txt --user
+          # Install any custom versions
+          if [ -d "hsve" ]; then
+            echo "Found custom howso-validator-enterprise version; installing..."
+            pip uninstall howso-validator-enterprise -y
+            pip install hsve/*.whl
+          fi
+          if [ -d "hsv" ]; then
+            echo "Found custom howso-validator version; installing..."
+            pip uninstall howso-validator -y
+            pip install hsv/*.whl
+          fi
+          if [ -d "synth" ]; then
+            echo "Found custom howso-synthesizer version; installing..."
+            pip uninstall howso-synthesizer -y
+            pip install synth/*.whl
+          fi
+          if [ -d "hse" ]; then
+            echo "Found custom howso-engine version; installing..."
+            pip uninstall howso-engine -y
+            pip install hse/*.whl
+          fi
+          if [ -d "amlg" ]; then
+            echo "Found custom amalgam-lang version; installing..."
+            pip uninstall amalgam-lang -y
+            pip install amlg/*.whl
+          fi
```

Comment:
Same as with the downloads, these could probably be put in a loop if a correctly data dict was constructed up front.